### PR TITLE
feat: add support to multiple dn

### PR DESCRIPTION
**Pull request by jmbass**
By adding support to semi-colon separated userToDnFormats, more directory branches can be targeted for one username.

### CODE REVIEW

**Comment by itoumlilt** [comment]
PR LGTM, but I don't have enough context about the subject so I would prefer a second pair of eyes before merge.

A small remark regarding Java best practice, we usually avoid having nested try-catch as finally block is usually for resource cleanup, and finally block exception can mask the one thrown in try-catch blocks, making debugging more complex.

**Comment by jmbass** [comment]
> PR LGTM, but I don't have enough context about the subject so I would prefer a second pair of eyes before merge.
> 
> A small remark regarding Java best practice, we usually avoid having nested try-catch as finally block is usually for resource cleanup, and finally block exception can mask the one thrown in try-catch blocks, making debugging more complex.

Hi @itoumlilt thanks for the flag! I've refactored it to prevent nested try-catch statements

**Review by itoumlilt — APPROVED** [review]

**Pull request merged**
